aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    adapter_common::{PreprocessedTransaction, VMAdapter},
+    aptos_vm::AptosVM,
+    block_executor::{storage_wrapper::VersionedView, AptosTransactionOutput},
+    data_cache::{AsMoveResolver, StateViewCache, StorageAdapter},
+    logging::AdapterLogSchema,
+    move_vm_ext::MoveResolverExt,
+};
+use aptos_aggregator::{delta_change_set::DeltaChangeSet, transaction::TransactionOutputExt};
+use aptos_block_executor::{
+    executor::MVHashMapView,
+    task::{ExecutionStatus, ExecutorTask},
+};
+use aptos_logger::prelude::*;
+use aptos_state_view::StateView;
+use aptos_types::{state_store::state_key::StateKey, write_set::WriteOp};
+use move_core_types::{
+    ident_str,
+    language_storage::{ModuleId, CORE_CODE_ADDRESS},
+    vm_status::VMStatus,
+};
+use std::collections::btree_map::BTreeMap;
+
+pub(crate) struct AptosVMWrapper<'a, S> {
+    vm: AptosVM,
+    base_view: &'a S,
```

Comment:
This `S` is always a `StateView`, right? Then this is really strange and the need for this wrapper type just demonstrates how broken the AptosVM is (and requires a larger overhaul). Because if you look at `AptosVM`, you see that it wraps `AptosVMImpl`, which in turn is created from a state view. That state view is then hidden inside of the storage adapter.  The need to be able to get hand a StateView already created acrobatic code in other places.

This is probably another set of PRs after this one, but I really wish we could drastically simplify the architecture here:

- Only one `AptosVM` (no VMImpl, BlockVM, MoveVmExt, and other complications)
- That `AptosVM` implements all the parallel and sequential execution logic. Multiple `impl AptosVM` split over files can help to tam the complexity.

Because parallel execution is build into our Move adapter there is really no need to maintain a code layer underneath without PE.